### PR TITLE
Yaw timeout for VTOL forwards transition

### LINKED ISSUE #3728: Implement timeout on yaw_reached for waypoints
Implement MIS_YAW_TMT // timeout for accepting waypoint yaw reached in seconds

Disabled: -1 // default
Instant: 0
VTOL: 5 // default

Timeout should not occur when next mission item is a VTOL transition

Lorenz wrote:
@AndreasAntener @sanderux on some waypoints other than takeoff and landing the commanded yaw heading might not be reachable. On some occasions (normal waypoints) we want to weathervane. But on other occasions (aligning before performing a transition) that would be an issue.

After thinking for a bit I would propose to weather-vane for all waypoints expect for a waypoint when followed by a transition. Failure to align should result in a transition abort there.

What do you think?

### PR DESCRIPTION
**Pull request by AndreasAntener**
This adds a yaw timeout on accepting the waypoint for the VTOL forwards transition because a VTOL might not reach yaw in wind. Fixes #3728

MIS_YAW_TMT
- default: -1
- > 0: timeout active on waypoints which force heading, ignore heading for acceptance on normal waypoints

MIS_YAW_TMT set to 10 seconds for default VTOL configurations

### CODE REVIEW

**Comment by AndreasAntener** [comment]
@LorenzMeier @sanderux timeout and forcing the heading for a transition added. How does this look?

Already flew the timeout in real, new is the mission failsafe. If this is how we want it for the moment we can merge I think.

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -183,8 +192,18 @@ MissionBlock::is_mission_item_reached()
 			/* check yaw if defined only for rotary wing except takeoff */
 			float yaw_err = _wrap_pi(_mission_item.yaw - _navigator->get_global_position()->yaw);
 
-			if (fabsf(yaw_err) < 0.2f) { /* TODO: get rid of magic number */
-				_waypoint_yaw_reached = true;
+			if (fabsf(yaw_err) < math::radians(_param_yaw_err.get())
+					|| (_param_yaw_timeout.get() >= -FLT_EPSILON &&
+						now - _time_wp_reached >= (hrt_abstime)_param_yaw_timeout.get() * 1e6f)) {
```
Comment (julianoes) [inline src/modules/navigator/mission_block.cpp]: Why -FLT_EPSILON? Shouldn't it be +FLT_EPSILON?

Also, please comment! How is a reader supposed to grasp what you're trying to achieve with this if...

Comment (julianoes) [inline src/modules/navigator/mission_block.cpp]: Oh, I understand from reading the parameter. A timeout of 0 means, just continue.
As I said, please comment!

**Comment by julianoes** [comment]
Reviewed, seems good other than the comment.

**Comment by AndreasAntener** [comment]
Thanks @julianoes , I will push an update in a sec that also takes care of setting the failure state properly.

**Comment by AndreasAntener** [comment]
Rebased on master

**Comment by AndreasAntener** [comment]
Just found one thing missing here, need to do one more commit.

**Comment by AndreasAntener** [comment]
Anybody knows why px4-stm32f4discovery_default fails on Travis?

**Comment by AndreasAntener** [comment]
Merged after SITL tests. @sanderux this will simplify testing this afternoon. Roman and I will rebase the TECS init, and also get the Kangaroo in before you go testing.

**Pull request merged**